enhancements/testdata-separation.md

```diff
@@ -0,0 +1,109 @@
+
+### Goals
+
+The goal is to create a solution which allows developers to continue to access and utilize the testdata while still having the option to ship an operator without said testdata in the bundle.
+
+### Non-Goals
+
+The goal of this EP is not to change the contents of the bundle nor the testdata, but just to reorganize them, either with multiple bundles or by allowing the tools to modify a bundle to achieve the desired state.
+
+## Proposal
+
+Please note that this proposed solution was elevated from the Alternatives selection after discussions, and the previous implementation has been moved down.
+
+Hard Split
+* Split existing bundle in a testdata only bundle paired with a operator essentials only bundle
+  * Allows the bundle to save significant space/reduce its size
+  * Allows operator developers to securely ship with including potentially sensitive testdata
+    * Users can use elevated permissions in testing without concerns about shipping those configs
+* Primary impact lies within scorecard internals, as well as scaffolding
+  * Scorecard needs logic to understand whether or not there are two bundle or one
+    * Recombine bundles at scorecard runtime
+      * Scorecard runs continue to behave the same, all this forces is a pre-processing step
+    * Allows scorecard test developers to contiue to use `/bundle` context
+* Allow discovery of test bundle using Annotation path
+  * Scorecard currently assumes testdata in the in current bundle, but can read the annotation to find another bundle
+    * This bundle can be configured as the seperate test bundle
+  * Seamless tranisition for users, exisiting CLI commands will continue to work in scipts if annotations are correctly configured
+  * This method does require manual user updates for configuration
+* Scorecard CLI extension
+  * Add ability to process one OR MORE bundles in the command line args
+  * If multiple args exist combine bundles and carry on
+* Opt In design
+  * By default bundles will continue to be scaffolded as is
+  * Users can always re-scaffold the bundle to toggle between types
+* Minimizes User Impact
+  * Current operators will still work
+  * New single bundle opertaors will not require any changes in the development or test process
+    * Those who opt in will have minimal manual adjustments needed for the 2 bundles to work together
+  * Scorecard test developers can expect the same scorecard API's and bundle mounting
+ 
+
+## Design Details
+
+### Test Plan
+
+For testing, ideally all new paths created in the CLI should be verfied in integration testing. Creating a full bundle, slim bundle, and then adding and/or removing testdata scaffolds from both. 
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+Implementation and testing of this feature should be sufficient for graduation. The intended implementation will maintain current user behavior as default.
+
```

Comment:
Can we add a section that discusses the impact for disconnected clusters/users? Since bundles are required to be mirrored to enable OLM to deploy them in disconnected clusters, the testdata (for better or worse) comes along for the ride.

There are a few concerns to work through if we split up the testdata from the operator portion of the bundle:
1. How do we ensure that testdata-only bundles are included in the mirrored set of bundles? We need the testdata-only bundle to show up in the CSVs relatedImages field, right?
2. If operator authors choose not to ship their testdata-only bundles, are they expected to leave the operator bundle's test annotations out?